src/workerd/jsg/jsg.h

```diff
@@ -2878,6 +2883,15 @@ class Lock {
   kj::Maybe<JsObject> resolveModule(
       kj::StringPtr specifier, RequireEsm requireEsm = RequireEsm::NO);
 
+  // Returns the capnp::SchemaLoader for this isolate/context
+  template <typename T>
+  const capnp::SchemaLoader& getCapnpSchemaLoader() const {
+    return KJ_ASSERT_NONNULL(
+        jsg::getAlignedPointerFromEmbedderData<T>(
```

Comment:
Why is this templated on T?
Couldn't you simply getAlignedPointerFromEmbedderData<ModuleRegistry> ?

Reply:
The `T` here is the global (e.g. `ServiceWorkerGlobalScope`) not `ModuleRegistryBase`... note that we're grabbing the `ContextPointerSlot::GLOBAL_WRAPPER` here, not the `ContetPointerSlot::MODULE_REGISTRY`.